### PR TITLE
Sscs 7806 reinstatement requests 2

### PR DESCRIPTION
**Pull request by danhamilton**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/mya/EvidenceUploadTest.java [hunk 1]
```diff
@@ -14,9 +14,11 @@
 public class EvidenceUploadTest extends BaseFunctionTest {
 
     @Test
-    public void uploadThenSubmitEvidenceToAppeal() throws IOException, JSONException {
+    public void uploadThenSubmitEvidenceToAppeal() throws IOException, JSONException, InterruptedException {
         CreatedCcdCase createdCcdCase = createCase();
 
+        Thread.sleep(5000L);
```
Comment (ryandac) [inline src/e2e/java/uk/gov/hmcts/reform/sscs/functional/mya/EvidenceUploadTest.java:20]: Is this needed?

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceAboutToSubmitHandler.java [hunk 2]
```diff
@@ -151,7 +167,12 @@ private void buildSscsDocumentFromScan(SscsCaseData sscsCaseData, State caseStat
                     List<SscsDocument> documents = new ArrayList<>();
 
                     if (!equalsIgnoreCase(scannedDocument.getValue().getType(), COVERSHEET)) {
+
                         SscsDocument sscsDocument = buildSscsDocument(sscsCaseData, scannedDocument, caseState);
+
+                        if (reinstatementFeatureFlag && REINSTATEMENT_REQUEST.getValue().equals(sscsDocument.getValue().getDocumentType())) {
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceAboutToSubmitHandler.java:173]: Is reinstatementFeatureFlag needed here?

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceAboutToSubmitHandlerTest.java [hunk 3]
```diff
@@ -49,6 +52,8 @@
 import uk.gov.hmcts.reform.sscs.ccd.domain.SscsDocumentDetails;
 import uk.gov.hmcts.reform.sscs.ccd.domain.SscsDocumentTranslationStatus;
 import uk.gov.hmcts.reform.sscs.ccd.domain.State;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.InterlocReviewState;
+
```
Comment (ryandac) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceAboutToSubmitHandlerTest.java:56]: looks like there is an unwanted new line in the import section

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/mya/EvidenceUploadTest.java (hunk 1 above)
Comment (danhamilton) [reply src/e2e/java/uk/gov/hmcts/reform/sscs/functional/mya/EvidenceUploadTest.java:20]: It's reduced Flakiness

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceAboutToSubmitHandler.java (hunk 2 above)
Comment (danhamilton) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceAboutToSubmitHandler.java:173]: Not strictly but keeps state from being polluted

**Review by ryandac — APPROVED** [review]

**Pull request merged**